mobile_verifier/src/verifier.rs

```diff
@@ -1,148 +1,223 @@
-    speedtests::{FetchError, SpeedtestAverages, SpeedtestRollingAverage, SpeedtestStore},
+    speedtests::{SpeedtestAverages, SpeedtestRollingAverage},
 };
 use anyhow::bail;
-use chrono::{DateTime, Duration, TimeZone, Timelike, Utc};
+use chrono::{DateTime, Duration, TimeZone, Utc};
 use db_store::meta;
-use file_store::{file_sink::FileSinkClient, traits::TimestampEncode, FileStore};
-use futures::{stream::Stream, StreamExt};
+use file_store::{
+    file_info_poller::FileInfoStream, file_sink::FileSinkClient,
+    heartbeat::CellHeartbeatIngestReport, speedtest::CellSpeedtestIngestReport,
+    traits::TimestampEncode, FileStore,
+};
+use futures::StreamExt;
 use helium_proto::RewardManifest;
-use mobile_config::{client::ClientError, Client};
+use mobile_config::Client;
 use price::PriceTracker;
+use retainer::Cache;
 use rust_decimal::{prelude::ToPrimitive, Decimal};
 use rust_decimal_macros::dec;
 use sqlx::{PgExecutor, Pool, Postgres};
-use std::{
-    collections::{hash_map::Entry, HashMap},
-    ops::Range,
+use std::{ops::Range, pin::pin, sync::Arc};
+use tokio::{
+    sync::mpsc::Receiver,
+    task::{JoinError, JoinHandle},
+    time::sleep,
 };
-use tokio::pin;
-use tokio::time::sleep;
 
 pub struct VerifierDaemon {
     pub pool: Pool<Postgres>,
-    pub heartbeats: FileSinkClient,
-    pub speedtest_avgs: FileSinkClient,
-    pub radio_rewards: FileSinkClient,
+    pub heartbeats: Receiver<FileInfoStream<CellHeartbeatIngestReport>>,
+    pub speedtests: Receiver<FileInfoStream<CellSpeedtestIngestReport>>,
+    pub valid_heartbeats: FileSinkClient,
+    pub valid_speedtests: FileSinkClient,
     pub mobile_rewards: FileSinkClient,
     pub reward_manifests: FileSinkClient,
+    pub data_transfer_ingest: FileStore,
     pub reward_period_hours: i64,
-    pub verifications_per_period: i32,
-    pub verification_offset: Duration,
-    pub verifier: Verifier,
+    pub config_client: Client,
     pub price_tracker: PriceTracker,
-    pub data_transfer_ingest: FileStore,
 }
 
 impl VerifierDaemon {
-    pub async fn run(mut self, shutdown: &triggered::Listener) -> anyhow::Result<()> {
+    pub async fn run(self, shutdown: &triggered::Listener) -> anyhow::Result<()> {
         tracing::info!("Starting verifier service");
 
-        let reward_period_length = Duration::hours(self.reward_period_hours);
-        let verification_period_length = reward_period_length / self.verifications_per_period;
+        let Self {
+            pool,
+            mut heartbeats,
+            mut speedtests,
+            valid_heartbeats,
+            valid_speedtests,
+            config_client,
+            mobile_rewards,
+            reward_manifests,
+            price_tracker,
+            data_transfer_ingest,
+            reward_period_hours,
+        } = self;
+
+        let heartbeat_pool = pool.clone();
+        let heartbeat_config_client = config_client.clone();
+        let mut heartbeat_verification_task: JoinHandle<anyhow::Result<()>> =
```

Comment:
Would it not be better structure to break each task out to its own module, ie a heartbeat loader, a speedtest loader, a rewarder.  Grouping them all into a single module makes the code somewhat more difficult to follow and maintain